src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets

```diff
@@ -321,6 +321,12 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
     <UpToDateCheckBuilt Include="@(RazorIntermediateAssembly)"
       Condition="'$(RazorCompileOnBuild)'=='true' AND '@(Content->WithMetadataValue('Extension', '.cshtml'))' != ''" />
+
+    <!-- Set up watchers for dotnet-watch -->
+    <Watch Include="@(Content->WithMetadataValue('Extension', '.razor'))" />
+
+    <Watch Include="@(Content->WithMetadataValue('Extension', '.cshtml'))"
+      Condition="'$(_Targeting30OrNewerRazorLangVersion)' == 'true' AND '$(AddCshtmlFilesToDotNetWatchList)' != 'false'" />
```

Comment:
MSBuild does not allow filtering by the absence of metadata in global evaluation. I cannot do 

``%(Content.Watch)` != false` or something along those lines that the runtime compilation package sets up. `AddCshtmlFilesToDotNetWatchList` is the next best thing

Reply:
Does `AddCshtmlFilesToDotNetWatchList` exist today? If not, i'd just declare it with a default value above. That way we could just do `'$(AddCshtmlFilesToDotNetWatchList)' == 'true'`

Reply:
This is fairly common pattern in MSBuild. It's more akin to saying check for existence of the property rather than declaring it anew here.